### PR TITLE
Use Findex v0.7.0

### PR DESCRIPTION
**Pull request by tbrezot**
Use Findex v0.7.0 interfaces.

### CODE REVIEW

File: src/crypto/sse/findex/interfaces/findex.ts [hunk 1]
```diff
@@ -55,20 +93,26 @@ export class Findex {
    * @param {Uint8Array} label used to create all indexes and to refresh them later on
    * @param {string[]} words a list of words
    * @param {number} loopIterationLimit this number helps to limit the number of results in a search query when unchaining Index Table Entry item
+   * @param graphRecursionLimit
+   * @param progress
    * @returns {Uint8Array[]} a list of Indexed Values
    */
   public async search(
     masterKeys: MasterKeys,
```
Comment (Manuthor) [inline src/crypto/sse/findex/interfaces/findex.ts:101]:
```suggestion
    searchKey: Uint8Array,
```

**Review by Manuthor — COMMENTED** [review]
great job!! 
please could you also:
- update cosmian_findex in package.json to 0.7.0
- bump version to 3.0.0 in package.json
- add the 3.0.0 version description in changelog.md

File: tests/crypto/sse/findex/implementations/common/findex_demo.ts [hunk 2]
```diff
@@ -79,14 +100,18 @@ export class FindexDemo extends Findex {
...
    */
   async searchWithLogicalSwitch(
     masterKeysFindex: MasterKeys,
     label: string,
     words: string,
     logicalSwitch: boolean,
-    loopIterationLimit: number
+    loopIterationLimit: number,
+    graphRecursionLimit: number,
+    progress: Function
```
Comment (ThibaudDauce) [inline tests/crypto/sse/findex/implementations/common/findex_demo.ts:114]: Add trailing comma?

File: tests/crypto/sse/findex/implementations/redis/tests/cloudproof.test.ts [hunk 3]
```diff
@@ -59,7 +59,12 @@ test("upsert+search", async () => {
       LABEL,
       "france spain",
       false,
-      1000
+      1000,
+      1000,
+      (res: Uint8Array[]) => {
+        console.log(res);
```
Comment (ThibaudDauce) [inline tests/crypto/sse/findex/implementations/redis/tests/cloudproof.test.ts:65]: Maybe add a test to check if the `progress` callback receive all the element by incrementing a counter and comparing with the final result?

File: tests/crypto/sse/findex/implementations/postgrest/tests/findex.test.ts [hunk 4]
```diff
@@ -23,6 +23,11 @@ test("upsert+search", async () => {
   expect(entries.length).toBe(577);
   expect(chains.length).toBe(792);
 
+  const progress = (res: Uint8Array[]) => {
+    console.log(res);
```
Comment (ThibaudDauce) [inline tests/crypto/sse/findex/implementations/postgrest/tests/findex.test.ts:27]: Maybe add a test to check if the `progress` callback receive all the element by incrementing a counter and comparing with the final result?

File: tests/crypto/sse/findex/implementations/redis/tests/findex.test.ts [hunk 5]
```diff
@@ -33,7 +33,62 @@ test("upsert+search", async () => {
       LABEL,
       "france",
       false,
-      1000
+      1000,
+      1000,
+      (res: Uint8Array[]) => {
+        console.log(res);
```
Comment (ThibaudDauce) [inline tests/crypto/sse/findex/implementations/redis/tests/findex.test.ts:39]: Maybe add a test to check if the `progress` callback receive all the element by incrementing a counter and comparing with the final result?

File: tests/crypto/sse/findex/implementations/common/findex_demo.ts (hunk 2 above)
Comment (tbrezot) [reply tests/crypto/sse/findex/implementations/common/findex_demo.ts:114]: eslint is managing this I think

**Pull request merged**